src/actions/authActions.jsx

```diff
@@ -136,3 +136,134 @@ export const googleAuth = () => {
       });
   };
 };
+
+export const updateProfile = (profileInfo) => {
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firestore = getFirestore();
+    firestore
+      .collection("profiles")
+      .doc(profileInfo.userId)
+      .update(profileInfo)
+      .then(() => {
+        dispatch({ type: "UPDATE_SUCCESS" });
+      })
+      .catch((err) => {
+        dispatch({ type: "UPDATE_ERROR", err });
+      });
+  };
+};
+
+export const deleteProfileAuth = () => {
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firebase = getFirebase();
+    const user = firebase.auth().currentUser;
+    user
+      .delete()
+      .then(() => {
+        dispatch({ type: "PROFILE_AUTH_DELETE_SUCCESS" });
+      })
+      .catch((err) => {
+        dispatch({ type: "PROFILE_AUTH_DELETE_ERROR", err });
+      });
+  };
+};
+
+export const deleteProfileData = (userId) => {
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firestore = getFirestore();
+    firestore
+      .collection("profiles")
+      .doc(userId)
+      .delete()
+      .then(() => {
+        dispatch({ type: "PROFILE_DATA_DELETE_SUCCESS" });
```

Comment:
these types should all be moved out to a variable and used in both the redux handler and here. the reason if you never want to be using magic strings -- all human beings make typos and it will break your code

google magic strings

also take a look at the very beginning of this redux example:
https://redux.js.org/basics/actions
```
const ADD_TODO = 'ADD_TODO'
```